### PR TITLE
Remove ba logger

### PR DESCRIPTION
**Pull request by MarioIvancik**
Ba logger can cause problems in this project. All logging should be handled one level higher. It's possible to detect long delays now due to newly added return values to functions that can cause delays.

## Summary by CodeRabbit

- **New Features**
  - Updated `FleetApiClient` methods to return additional context on request delays, enhancing usability.
  - Improved `handleDelays` method to indicate success or failure of delay handling.

- **Bug Fixes**
  - Removed logging setup in tests to streamline execution, though this may impact debugging capabilities.

- **Documentation**
  - Updated README to reflect the removal of the `bringauto-logger` dependency.

- **Refactor**
  - Enhanced return types and const-correctness across various methods in `FleetApiClient` and `RequestFrequencyGuard`.

- **Tests**
  - Updated tests for `FleetApiClient` to validate delay handling.
  - Improved assertions in `RequestFrequencyGuardTests` for better outcome verification.

### CODE REVIEW

File: src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp [hunk 1]
```diff
@@ -42,77 +42,85 @@ FleetApiClient::FleetApiClient(
...
 	deviceIdPtr_->setModuleId(deviceIdentification.module);
 	deviceIdPtr_->setType(deviceIdentification.device_type);
 	deviceIdPtr_->setRole(std::string(static_cast<char*>(deviceIdentification.device_role.data)));
 	deviceIdPtr_->setName(std::string(static_cast<char*>(deviceIdentification.device_name.data)));
 }
 
-
-std::vector<std::shared_ptr<model::Car>> FleetApiClient::getCars(std::optional<int64_t> since, std::optional<bool> wait) {
-	auto carsRequest = carApi_->availableCars(wait.value_or(false), since.value_or(0));
+std::pair<std::vector<std::shared_ptr<model::Car>>, bool> FleetApiClient::getCars(
```
Comment (koudis) [inline src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp]: Please do not use Bool. Use enum class. What if another state needs to be returned?

Maybe create one Enum class for the FleetApiClient...

Comment (MarioIvancik) [reply src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp]: Enum for return codes added

File: include/bringauto/fleet_protocol/http_client/FleetApiClient.hpp [hunk 2]
```diff
@@ -22,6 +22,14 @@ class FleetApiClient {
 		STATUS_ERROR
 	};
 
+	/**
+	 * @brief Enum class for return codes
+	 */
+	enum class ReturnCode {
+		OK,
+		DELAYED
```
Comment (koudis) [inline include/bringauto/fleet_protocol/http_client/FleetApiClient.hpp]: What does it mean delayed? If someone reads the doc it is not clear what doe it means.

Comment (MarioIvancik) [reply include/bringauto/fleet_protocol/http_client/FleetApiClient.hpp]: added an explanation

**Review by koudis — APPROVED** [review]

**Pull request merged**